src/chains/ethereum/ethereum/tests/api/eth/getBlockByNumber.test.ts

```diff
@@ -0,0 +1,99 @@
+import assert from "assert";
+import EthereumProvider from "../../../src/provider";
+import getProvider from "../../helpers/getProvider";
+import compile from "../../helpers/compile";
+const DEFAULT_DIFFICULTY = 1;
+let provider: EthereumProvider;
+
+// helper to mine an arbitrary number of blocks using dummy transactions
+const mineBlocks = async (number: number, provider: EthereumProvider) => {
+  const [from] = await provider.send("eth_accounts");
+  for (let x = number; x > 0; x--) {
+    await provider.send("eth_subscribe", ["newHeads"]);
+
+    await provider.send("eth_sendTransaction", [{ from, to: from }]);
+
+    await provider.once("message");
+  }
+};
+
+describe("api", () => {
+  describe("eth", () => {
+    describe("getBlockByNumber", () => {
+      describe("difficulty", () => {
+        beforeEach(async () => {
+          provider = await getProvider();
+        });
+
+        afterEach(async () => {
+          provider && (await provider.disconnect());
+        });
+        it("returns the block difficulty", async () => {
+          const block = await provider.send("eth_getBlockByNumber", ["latest"]);
+          assert.strictEqual(block.difficulty, `0x${DEFAULT_DIFFICULTY}`);
+        });
+      });
+
+      describe("totalDifficulty", () => {
+        describe("when the default values are used", () => {
+          beforeEach(async () => {
+            provider = await getProvider();
+          });
+          afterEach(async () => {
+            provider && (await provider.disconnect());
+          });
+
+          it("equals the block difficulty for the genesis block", async () => {
+            const block = await provider.send("eth_getBlockByNumber", ["0x0"]);
+            assert.strictEqual(
+              block.totalDifficulty,
+              `0x${DEFAULT_DIFFICULTY}`
```

Comment:
If `DEFAULT_DIFFICULTY` is ever greater than 9 in the future this test will fail.

Other ways of solving this:

```
assert.strictEqual(
  block.totalDifficulty,
  Quantity.from(DEFAULT_DIFFICULTY).toString()
);
```

or 

```
assert.strictEqual(
  parseInt(block.totalDifficulty, 16),
  DEFAULT_DIFFICULTY
);
```

or

```
assert.strictEqual(
  BigInt(block.totalDifficulty),
  DEFAULT_DIFFICULTY // change DEFAULT_DIFFICULTY to a BigInt
);
```

and lots of others, I'm sure.

Reply:
Can you also update other places that use the `0x{number}` pattern in these tests? Thanks!

Reply:
Sorry, I meant to update these but must have overlooked them. Updated!